modules/claims_api/lib/custom_error.rb

```diff
@@ -8,84 +8,53 @@ def initialize(error, claim, method)
-         @error.is_a?(Faraday::NilStatusError) || @error.is_a?(Faraday::TimeoutError) ||
-         @error.is_a?(::Common::Exceptions::BackendServiceException) ||
-         @error.is_a?(::Common::Exceptions::ExternalServerInternalServerError) ||
-         @error.is_a?(::Common::Exceptions::BadGateway) || @error.is_a?(Faraday::SSLError) ||
-         @error.is_a?(Faraday::ServerError)
-        errors = { errors: [{ 'title' => 'Service Exception',
-                              'key' => @source,
-                              'detail' => 'A re-tryable error has occurred, original_error: ' \
-                                          "#{@error}.", status: @status, code: @code }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::ServiceError, errors
-
-      elsif @error.is_a?(StandardError) || @error.is_a?(Faraday::BadRequestError) ||
-            @error.is_a?(Faraday::ConflictError) || @error.is_a?(Faraday::ForbiddenError) ||
-            @error.is_a?(Faraday::ProxyAuthError) || @error.is_a?(Faraday::ResourceNotFound) ||
-            @error.is_a?(Faraday::UnauthorizedError) || @error.is_a?(Faraday::UnprocessableEntityError) ||
-            @error.is_a?(Faraday::ClientError)
-
-        errors = { errors: [{ 'title' => 'Client error',
-                              'key' => @source,
-                              'detail' => 'A client exception has occurred, job will not be re-tried. ' \
-                                          "original_error: #{@error}.", status: '400', code: '400' }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::BadRequest, errors
+    def build_error
+      case @error
+      when Faraday::ParsingError
+        raise_backend_exception(@error.class, @error, 'EVSS502')
+      when ::Common::Exceptions::BackendServiceException
+        raise ::Common::Exceptions::Forbidden if @error&.original_status == 403
+
+        raise_backend_exception(@error.class, @error, 'EVSS400') if @error&.original_status == 400
+        raise ::Common::Exceptions::Unauthorized if @error&.original_status == 401
       else
-        errors = { errors: [{ 'title' => 'Unknown error',
-                              'key' => @source,
-                              'detail' => 'An unknown error has occurred, and the custom_error file may ' \
-                                          "need to be modified. original_error: #{@error}.", status: @status,
-                              code: @code }] }
-        log_outcome_for_claims_api(errors)
-        raise ::Common::Exceptions::ServiceError, errors
+        raise @error
       end
     end
 
     private
 
-    def log_outcome_for_claims_api(errors)
-      ClaimsApi::Logger.log('526_docker_container',
-                            detail: "claims_api-526-#{@method},  errors: #{errors}", claim: @claim&.id)
-    end
+    def raise_backend_exception(source, error, key = 'EVSS')
+      error_details = get_error_details(error)
```

Comment:
I think this is redundant logic. we can just pass `e.original_body` to line :34 and get the same result.

```
(byebug) @error.original_body
{:messages=>[{:key=>"header.va_eauth_birlsfilenumber.Invalid", :severity=>"ERROR", :text=>"Size must be between 8 and 9"}]}
   30:       error_details = get_error_details(error)
   31:
=> 32:       raise ::Common::Exceptions::BackendServiceException.new(
   33:         key,
   34:         { source: },
   35:         error&.original_status,
   36:         error_details
(byebug) error_details
[{:key=>"header.va_eauth_birlsfilenumber.Invalid", :severity=>"ERROR", :text=>"Size must be between 8 and 9"}]
```

actually this throws an error in the validate. [without the `:messages`](https://github.com/department-of-veterans-affairs/vets-api/blob/4e7c4d2893081bfbe6b6cca66f0bbb65fef6cf4b/modules/claims_api/app/controllers/claims_api/v1/forms/disability_compensation_controller.rb#L180).

```
            "code": "500",
            "status": "500",
            "meta": {
                "exception": "no implicit conversion of Symbol into Integer",
```

```
=> 180:           error_details = e&.original_body&.[](:messages)
   181:           raise ::Common::Exceptions::UnprocessableEntity.new(errors: format_526_errors(error_details))
   182:         rescue ::Common::Exceptions::GatewayTimeout,
   183:                ::Timeout::Error,
   184:                ::Faraday::TimeoutError,
(byebug) e&.original_body&.[](:messages)
*** TypeError Exception: no implicit conversion of Symbol into Integer

nil
```

if we pass the original body we can just change the `claim_establisher` to record `evss_response = e.messages`. unless you saw a differing format in the submit call?

Reply:
@FonzMP Good catch.  I think the fix might actually be to remove the line `error_details = e&.original_body&.[](:messages)` in the  V1 `disability_compensation_controller`. This was added when building the V2 validate service because of the different format from the docker container.  I think the current update renders this unnecessary now.  The error here and in the V1 `claims_controller` should be in the same format I _believe_, so not having `:messages` over there ([v1 claims_controller formatter ](https://github.com/department-of-veterans-affairs/vets-api/pull/16197/files#diff-c8c2cbea22f56700c7ecefe8881aabb41b0e127a62c1bd0c7a772696a9193fecR66)) would mean we no longer want to account for this over here too I suspect.  Thoughts?

Reply:
still don't know if I'm quite following because the docker response is the same for v1 and v2. no formatting has been changed. plus the below logic for `error_details` just extracts the `:messages` we get from the container? Anywho, it behaves in the same manner, so 🤷 

```
(byebug) error.original_body
{:messages=>[{:key=>"header.va_eauth_birlsfilenumber.Invalid", :severity=>"ERROR", :text=>"Size must be between 8 and 9"}]}
(byebug) error_details
[{:key=>"header.va_eauth_birlsfilenumber.Invalid", :severity=>"ERROR", :text=>"Size must be between 8 and 9"}]
```

Reply:
I think I see where you're driving that.. it's because of the loss of the ability to call `e.messages` from the former error.

[Service Exception being raised](https://github.com/department-of-veterans-affairs/vets-api/blob/86b38e2bb97a03d4ce24651b3e485edeb14c52b7/lib/evss/disability_compensation_form/service.rb#L105)
[Error class](https://github.com/department-of-veterans-affairs/vets-api/blob/master/lib/evss/disability_compensation_form/service_exception.rb)
[parent giving us the goods](https://github.com/department-of-veterans-affairs/vets-api/blob/86b38e2bb97a03d4ce24651b3e485edeb14c52b7/lib/evss/service_exception.rb#L10)